Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -580,29 +640,238 @@ HRESULT STDMETHODCALLTYPE CreateStreamFromKey(_In_ const void* fontFileReference
         RETURN_HR_IF_NULL(E_INVALIDARG, fontFileStream);
 
         ComPtr<DWriteFontBinaryDataStream> ret;
-        RETURN_IF_FAILED(MakeAndInitialize<DWriteFontBinaryDataStream>(&ret, _dataProvider.get()));
+        RETURN_IF_FAILED(MakeAndInitialize<DWriteFontBinaryDataStream>(&ret, _data.get()));
 
         *fontFileStream = ret.Detach();
         return S_OK;
     }
 
 private:
-    woc::unique_cf<CGDataProviderRef> _dataProvider;
+    woc::unique_cf<CFDataRef> _data;
+};
+
+#pragma region Font File Registration
+// Implementation of IDwriteFontFileEnumerator, which enumerates over internal data and returns a font file
+class DWriteFontFileEnumerator : public RuntimeClass<RuntimeClassFlags<WinRtClassicComMix>, IDWriteFontFileEnumerator> {
+protected:
+    InspectableClass(L"Windows.Bridge.DirectWrite", TrustLevel::BaseTrust);
+
+public:
+    DWriteFontFileEnumerator() {
+    }
+
+    HRESULT RuntimeClassInitialize(CFArrayRef fontDatas, std::vector<ComPtr<IDWriteFontFile>>* previouslyCreatedFiles) {
+        CFRetain(fontDatas);
+        m_fontDatas.reset(fontDatas);
+        m_previouslyCreatedFiles = previouslyCreatedFiles;
+        return S_OK;
+    }
+
+    HRESULT STDMETHODCALLTYPE GetCurrentFontFile(_Out_ IDWriteFontFile** fontFile) {
+        if (location < m_previouslyCreatedFiles->size()) {
+            *fontFile = m_previouslyCreatedFiles->at(location).Get();
+        } else {
+            ComPtr<IDWriteFactory> dwriteFactory;
+            RETURN_IF_FAILED(DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED, __uuidof(IDWriteFactory), &dwriteFactory));
+
+            ComPtr<DWriteFontBinaryDataLoader> loader;
+            RETURN_IF_FAILED(
+                MakeAndInitialize<DWriteFontBinaryDataLoader>(&loader, (CFDataRef)CFArrayGetValueAtIndex(m_fontDatas.get(), location)));
+            RETURN_IF_FAILED(dwriteFactory->RegisterFontFileLoader(loader.Get()));
+
+            int unused;
+            RETURN_IF_FAILED(dwriteFactory->CreateCustomFontFileReference(&unused, sizeof(int), loader.Get(), fontFile));
+            m_previouslyCreatedFiles->emplace_back(*fontFile);
+        }
+        return S_OK;
+    }
+
+    HRESULT STDMETHODCALLTYPE MoveNext(_Out_ BOOL* hasCurrentFile) {
+        *hasCurrentFile = ++location < CFArrayGetCount(m_fontDatas.get());
+        return S_OK;
+    }
+
+private:
+    int location = -1;
```

Comment:
it's supposed to start before the position of the first element, which would be 0 #WontFix